src/smtencoding/FStar.SMTEncoding.Encode.fs

```diff
@@ -502,76 +502,67 @@ let encode_top_level_let :
       binders@extra_formals, body
     in
 
-    let destruct_bound_function flid t_norm e
-      : (S.binders    //arguments of the lambda abstraction
-      * S.term        //body of the lambda abstraction
-      //* S.binders     //arguments of the function type, length of this component is equal to the first
-      * S.comp)       //result comp
-      * bool          //if set, we should generate a curried application of f
+    let destruct_bound_function flid t e
+      : (S.binders    //arguments of the (possibly reified) lambda abstraction
+       * S.term       //body of the (possibly reified) lambda abstraction
+       * S.comp)      //result comp
+//       * bool         //if set, we should generate a curried application of f
     =
       (* The input type [t_norm] might contain reifiable computation type which must be reified at this point *)
-      let get_result_comp c =
-          if is_reifiable_comp env.tcenv c
-          then S.mk_Total <| reify_comp ({env.tcenv with lax = true}) c U_unknown
-          else c
-      in
+
+      let tcenv = {env.tcenv with lax=true} in
```

Comment:
I don't think `lax` makes a difference here. Can we just define `tcenv = env.tcenv`?